### DISCUSSION TITLE
Clarification on SSR

### DISCUSSION THREAD

**Discussion by LoiLock** [pmndrs/valtio#903] [General]
A while back, [someone said](https://github.com/pmndrs/valtio/issues/14#issuecomment-750923544) that Valtio, as it is, doesn't really work with SSR as the state between requests would get mixed up. Could someone clarify what is meant by that? Does it mean that user A could see the state requested by user B under certain circumstances?

For reference: (https://github.com/pmndrs/valtio/issues/14#issuecomment-750923544)

**Comment by dai-shi** [comment]
It's not precisely related to Valtio.
Any global variable is shared on the server across sessions.

```js
let globalCounter = 0;

const Foo = () => {
  const counter = ++globalCounter;
  return <p>count: {counter}</p>;
};
```
